google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageOptions.java

```diff
@@ -112,13 +126,28 @@ StorageSettings getStorageSettings() throws IOException {
             .setCredentialsProvider(credentialsProvider)
             .setClock(getClock());
 
+    InstantiatingGrpcChannelProvider.Builder channelProviderBuilder =
+        InstantiatingGrpcChannelProvider.newBuilder().setEndpoint(endpoint);
+
+    if (attemptDirectPath) {
+      channelProviderBuilder
+          .setAttemptDirectPath(true)
+          .setDirectPathServiceConfig(
+              ImmutableMap.of(
+                  "loadBalancingConfig",
+                  ImmutableList.of(
+                      ImmutableMap.of(
+                          "grpclb",
+                          ImmutableMap.of(
+                              "childPolicy",
+                              ImmutableList.of(
+                                  ImmutableMap.of("round_robin", ImmutableMap.of())))))));
```

Comment:
This is ugly, no offence but is there a better way to do this?

Reply:
Ended up not being needed. I removed it.